### ISSUE TITLE
Implement sending of emails

### ISSUE THREAD

**Issue by lache-melvin** [msupply-foundation/open-msupply#5821]
Using [Notify](https://github.com/msupply-foundation/notify/blob/main/backend/service/src/email/mod.rs) implementation as a reference, we need:

- [ ] Email service to process and try send all unsent emails
- [ ] Scheduled task to trigger this email process
- [ ] Mailhog set up for testing
- [ ] Add email configuration to app config (required only for OMS central)

**Comment by lache-melvin** [comment]
@mark-prins - can you please confirm if it okay to require email config on OMS central from 2.5?

Or do we need a way to hide feedback form on remote sites if email sending not set up?

**Comment by mark-prins** [comment]
if doing option 1 I'm unsure how we force the upgrade process to require configuring the email settings, aside from something critical like the server non-starting. Any thoughts on that?

Preference is for option 2 - that would mean the settings aren't required to be entered for our upgrades, but that feedback won't be unsent when the UI says that it will be. It means more work though, so I'm happy to require email config if we can swing the upgrades.

**Comment by noel-yeldos** [comment]
@lache-melvin @mark-prins  just had a conversation with @jmbrunskill on the back of this comment: https://github.com/msupply-foundation/open-msupply/pull/5816#pullrequestreview-2505174818
 
James has advised that we will need to implement 'exponential back-off' as per this issue: https://github.com/msupply-foundation/health-supply-hub/issues/350

A ```retry_at``` column will need to be added. It is currently implemented in Notify in a separate ```notification_events``` table but in OMS it will need to be in the ```email_queue``` table

**Comment by lache-melvin** [comment]
@mark-prins Could stop sync until central configuration is correct? Might be capability we want for other things in future too? Bit like the version mismatch error returned to the remote site, plus good logging on central server so we know what needs to be configured? And well documented for support doing upgrades.. 😅

**Comment by lache-melvin** [comment]
Epic thanks for capturing here @noel-yeldos! Yeah that would be great to get in here too 🔥

**Comment by lache-melvin** [comment]
> @mark-prins Could stop sync until central configuration is correct? Might be capability we want for other things in future too? Bit like the version mismatch error returned to the remote site, plus good logging on central server so we know what needs to be configured? And well documented for support doing upgrades.. 😅

I also know we are trying not to put more on support right now, so maybe need to sync "email sending now configured" out to remote sites and show contact form based on that. Email server being a blocker for upgrades might not be ideal right now?

Still think long term worth thinking about how we handle this, might need to take to discussion elsewhere though 😁

**Comment by jmbrunskill** [comment]
I think it's ok to just log errors about failing to send if miss configured, and make sure support is up to speed on the new configuration option.

If we want to try to prevent start up without configuration that's going to be difficult as OMS currently doesn't know if it's a Central Server or not until it's synced.

I like your idea about syncing a `email` setup now record but I think that's overkill (imho). Once we have global prefs we should probably hide this feature behind a global pref, then we can turn it on depending on if a customer actually wants this feature or not? And when turning it on we could have a note specifying that it requires email configuration.

**Comment by jmbrunskill** [comment]
Side issue, do we want to limit email sending to only the central server? Seems plausible we might want to send directly from a site in future if configured? But maybe should limit it to just Central for now?

**Comment by lache-melvin** [comment]
@jmbrunskill yeah wanting to limit to central only for now - think remote sites handling their own emails can be a future problem 😁 

Hehe ok I like the log an error and let them fix it approach 🚀

**Comment by mark-prins** [comment]
> Could error on v6 sync until central configuration is correct?

yes, but it sounds like you've moved on from that approach anyway! Happy to log an error on central as per @jmbrunskill's suggustion

**Comment by marthakeezy** [comment]
Tested as a part of feature release testing in #5884

Moving to Done ✅
